### PR TITLE
Upgrade NPM for os-packager + remove debugging code

### PR DESCRIPTION
**Pull request by akariv**

### CODE REVIEW

File: docker-config/os-packager/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,17 @@
...
+
 
 RUN apk add --update python
+RUN apk add --update --virtual=build-dependencies ca-certificates build-base
 RUN apk add git
 RUN apk add nodejs
 RUN git clone http://github.com/openspending/os-packager.git app
+RUN npm install npm -g
+RUN npm --version
 RUN cd app && npm install --only=prod
```
Comment (pwalsh) [inline docker-config/os-packager/Dockerfile]: Why are you separating the dev and production builds for npm? Just for debugging? Also, we should not ever need to install dev dependencies.

**Comment by pwalsh** [comment]
@akariv approved.

File: docker-config/os-packager/Dockerfile (hunk 1 above)
Comment (akariv) [inline docker-config/os-packager/Dockerfile]: TBH I never gave it too much thought... 
but now that I look at the package.json, I don't really follow the logic of the separation between dev and prod dependencies here: https://github.com/openspending/os-packager/blob/master/package.json

e.g. why is jquery & bootstrap a dev dep., but underscore and d3 not? or angular a dev dep., but angular-route not?

Perhaps @kravets-levko can shed some light here?

Comment (pwalsh) [inline docker-config/os-packager/Dockerfile]: Looks like a bit of tidying needs to be done there.

As we build on the server (compile our production assets) then I guess actually we do need the dev dependencies.....

I'm not sure how this is working here in packager but we've also been using npm's postinstall hooks to build assets, eg:

https://github.com/okfn/spend-publishing-dashboard/blob/master/package.json#L95

Comment (akariv) [inline docker-config/os-packager/Dockerfile]: I'm running gulp directly here (no hooks), so we do need that - although the test dependencies are probably not necessary.

**Pull request merged**